src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1547,5 +1547,24 @@ internal bool Equals(TypeSymbolWithAnnotations other)
         {
             throw ExceptionUtilities.Unreachable;
         }
+
+        public static bool Equals(TypeSymbol left, TypeSymbol right, TypeCompareKind comparison)
+        {
+            if (comparison == TypeCompareKind.ConsiderEverything && ((object)right == null))
+            {
+                return (object)left == (object)null;
+            }
+
+            return (comparison == TypeCompareKind.ConsiderEverything && (object)left == (object)right)
+                || right.Equals(left, comparison);
+        }
+
+        [Obsolete("Use TypeSymbol.Equals method.", true)]
+        public static bool operator ==(TypeSymbol left, TypeSymbol right)
```

Comment:
>public static bool operator ==(TypeSymbol left, TypeSymbol right) [](start = 8, length = 65)

Consider also adding obsolete operators:
```
public static bool operator ==(Symbol left, TypeSymbol right)
public static bool operator ==(TypeSymbol left, Symbol right)
public static bool operator ==(object left, TypeSymbol right)
public static bool operator ==(TypeSymbol left, object right)
```

Reply:
I'll add the ones for `Symbol`, but I found that the ones for `object` are rather annoying. 
We have code that does `(object)x == y`, which seems fine to leave as-is.

Reply:
The comparison between Symbol and TypeSymbol was used in a few places.
I ended up switching `basesBeingResolved` to use `TypeSymbol` to fix some of those (commit 7).

---
In reply to: [242663977](https://github.com/dotnet/roslyn/pull/31814#discussion_r242663977) [](ancestors = 242663977)